Dockerfile

```diff
@@ -17,8 +17,18 @@
 ################################################################################
 # Build
 FROM maven:3.8.4-openjdk-11 AS build
-WORKDIR /app
 
+WORKDIR /tmp/flink/plugins
+ENV FLINK_VERSION=1.14.3
+RUN --mount=type=cache,target=/root/.m2 \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-datadog:$FLINK_VERSION -DoutputDirectory=flink-metrics-datadog; \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-graphite:$FLINK_VERSION -DoutputDirectory=flink-metrics-graphite; \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-influxdb:$FLINK_VERSION -DoutputDirectory=flink-metrics-influxdb; \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-prometheus:$FLINK_VERSION -DoutputDirectory=flink-metrics-prometheus; \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-slf4j:$FLINK_VERSION -DoutputDirectory=flink-metrics-slf4j; \
+      mvn dependency:copy -Dartifact=org.apache.flink:flink-metrics-statsd:$FLINK_VERSION -DoutputDirectory=flink-metrics-statsd
+
```

Comment:
Do we really need all the reporters? But then, how to decide which one to include or not? I think this is fine and we can come back to it later. It would be really nice though if we can add a section to README how to customize the image, since that will be a frequent use case no matter what we decide here.